internal/Data/TypeRep/CacheMap.hs

```diff
@@ -70,32 +73,63 @@ fromAny = unsafeCoerce
 empty :: TypeRepMap f
 empty = TypeRepMap mempty mempty mempty
 
+-- | Construct a 'TypeRepMap' with a single element.
+one :: forall a f . Typeable a => f a -> TypeRepMap f
```

Comment:
There are many possible names for this function: `one`, `only`, `single`, `singleton`, etc. I think the choice should be made in favor of the most common one, `singleton` - it's used by both `containers` and `dependent-map`.

Reply:
@int-index I known that `singleton` is more common name. But I'd like to have shorter name. Though, it's minor, can be renamed.

Reply:
I don't like the name `singleton` anyway, so use your judgement. `one` is equally sensible, just not what I'm used to.